src/app/user/modules/records/components/record-form/record-form.component.ts

```diff
@@ -168,14 +212,40 @@ export class RecordFormComponent implements OnInit {
     };
   }
 
-  onSubmit() {
-    const req: Record = {
-      off_campus_event: this.buildOffCampusEvent(),
-      user: this.authService.userID,
-      contents: this.buildContents(),
-      attachments: this.attachments,
+  private buildOffCampusEventToUpdate(): OffCampusEvent {
+    const value = this.recordForm.value;
+    const offCampusEvent = this.record.off_campus_event as OffCampusEvent;
+    return {
+      id: offCampusEvent.id,
+      name: value.name,
+      time: value.time,
+      location: value.location,
+      num_hours: value.numHours,
+      num_participants: value.numParticipants,
     };
-    this.recordService.createOffCampusRecord(req).subscribe(
+  }
+
+  onSubmit() {
+    let targetRecord: Observable<Record>;
+    if (this.toUpdate) {
```

Comment:
Most of these initialization process for request is same, so we could construct the request first, then based on `isUpdateMode`, we choose different service method to send the request.